### PR TITLE
feat(search): cache glyph text search pages

### LINKED ISSUE #69: perf(search): locate_occurrences の探索アルゴリズムを改善する
## 背景
`locate_occurrences_with_strategy` は `Vec<char>` 化したテキストに対して線形スライス比較を行っている。文書・クエリが長い場合に処理コストが増えやすい。

- 対象: `src/search/engine.rs`
- 主な箇所: `locate_occurrences_with_strategy`

## 問題
- 現在は `search_chars[cursor..cursor+query_len] == query_chars[..]` を繰り返す実装。
- クエリ長が長いケースで比較コストが嵩む。

## 期待する改善
- 文字列探索をより効率的な手法へ置換する（例: `find` ベース + オフセット管理、もしくは適切な高速探索実装）。
- 既存仕様（非オーバーラップヒット、case sensitivity、空白無視戦略、スニペット/矩形抽出）を維持する。

## 受け入れ条件
- 検索結果の互換性を維持（既存テスト通過）。
- 長文ページまたは長いクエリで、探索処理時間が改善していることを確認できる。
- 可読性と保守性を損なわない実装になっている。

### PR DESCRIPTION
**Pull request by shiguri-01**
## Summary
- switch search to use the positioned glyph stream as the primary searchable text source
- keep glyphs searchable when their highlight geometry is unavailable, while surfacing the existing highlight warning
- run whitespace-insensitive matching only as a per-page fallback when normal matching finds no hits
- cache extracted positioned text pages by document/page with bounded LRU and memory limits
- replace occurrence lookup with string find-based matching

## Rationale
The local search corpus and performance notes showed that positioned text extraction dominates repeated searches, while searching already-extracted glyphs is much cheaper. Caching TextPage values lets later searches in the same PDF reuse extraction work.

The corpus also showed that XeLaTeX-style PDFs can drop word spaces entirely, so strict phrase search is too brittle. Page-level whitespace fallback preserves those matches without always paying the second-pass cost or preferring surprising no-space hits.

This also addresses the core request in #69 by replacing the old slice-comparison occurrence scan with str::find plus character/glyph offset mapping.

Closes #69.

## Verification
- cargo fmt --check
- cargo test
- cargo clippy --all-targets --all-features -- -D warnings

## Notes
The untracked tmp/ search corpus and reports were used as local validation material and are intentionally not included in this PR.

## Summary by CodeRabbit

* **New Features**
  * Search retries with a whitespace-insensitive substring matcher when direct matches fail.
  * Per-worker LRU memory-budget cache to speed repeated page searches and reuse extracted positioning.

* **Bug Fixes**
  * Better handling when glyph bounding information is missing; matches are still reported and highlight availability is flagged when geometry is unavailable.

* **Tests**
  * Expanded tests covering cache behavior, highlight availability, and matching edge cases.

* **Documentation**
  * Updated runtime search behavior docs.

### CODE REVIEW

File: src/search/engine.rs [hunk 1]
```diff
@@ -91,6 +94,87 @@ enum WorkerControl {
     Shutdown,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+struct SearchPageCacheKey {
+    doc_id: u64,
+    page: usize,
```
[the remark this answers is not in the document]
Comment (shiguri-01) [reply src/search/engine.rs:100]: Thanks, this is a valid risk in the abstract. I am going to leave this unchanged in this PR because the current document lifecycle does not reopen/replace a PDF within the same session while preserving the search worker cache, so there is no current path that would reuse stale positioned text for a replaced same-size file.

`doc_id` is also already the cache identity used by rendering and outline code, and changing it to a content hash or per-open UUID would be a broader cross-cache contract change. I would rather keep this PR scoped to glyph-search behavior and revisit document identity as a separate change when we add an explicit reload/reopen flow.

[the remark this answers is not in the document]
Comment (shiguri-01) [reply src/search/engine.rs:100]: Yes, please go ahead and open that issue.
